<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Linear Genomic Selection Indices ‚Ä¢ selection.index</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><script src="../extra.js"></script><meta property="og:title" content="Linear Genomic Selection Indices">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">selection.index</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started with selection.index</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-selection-index.html">The Linear Phenotypic Selection Index Theory</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-phenotypic-selection-indices.html">Constrained Linear Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-marker-selection-indices.html">Linear Marker and Genome-Wide Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-selection-indices.html">Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-genomic-selection-indices.html">Constrained Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-eigen-selection-index-methods.html">Linear Phenotypic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-eigen-selection-index-methods.html">Linear Genomic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/multistage-linear-selection-indices.html">Multistage Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/stochastic-simulation-linear-phenotypic-indices.html">Stochastic Simulation of Linear Phenotypic Selection Indices</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zankrut20/selection.index/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Linear Genomic Selection Indices</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zankrut20/selection.index/blob/master/vignettes/linear-genomic-selection-indices.Rmd" class="external-link"><code>vignettes/linear-genomic-selection-indices.Rmd</code></a></small>
      <div class="d-none name"><code>linear-genomic-selection-indices.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="abstract">Abstract<a class="anchor" aria-label="anchor" href="#abstract"></a>
</h2>
<p>The linear genomic selection index (LGSI) is a linear combination of
genomic estimated breeding values (GEBVs) used to predict the individual
net genetic merit and select individual candidates from a non-phenotyped
testing population as parents of the next selection cycle. In the LGSI,
phenotypic and marker data from the training population are fitted into
a statistical model to estimate all individual genome marker effects;
these estimates can then be used in subsequent selection cycles to
obtain GEBVs that are predictors of breeding values in a testing
population for which there is only marker information.</p>
<p>Applying the LGSI requires predicting and ranking the net genetic
merit of candidates for selection using only marker information. The
combined LGSI (CLGSI), conversely, uses both phenotypic and GEBV
information jointly to predict the net genetic merit. The CLGSI is
typically used in the training populations to estimate true merit.</p>
<p>This vignette covers calculating LGSI and CLGSI using the
<code>selection.index</code> package.</p>
</div>
<div class="section level2">
<h2 id="the-linear-genomic-selection-index">5.1 The Linear Genomic Selection Index<a class="anchor" aria-label="anchor" href="#the-linear-genomic-selection-index"></a>
</h2>
<div class="section level3">
<h3 id="basic-conditions-for-constructing-the-lgsi">Basic Conditions for Constructing the LGSI<a class="anchor" aria-label="anchor" href="#basic-conditions-for-constructing-the-lgsi"></a>
</h3>
<p>To construct a valid LGSI, four essential conditions must be met: 1.
All marker effects must be estimated simultaneously in the training
population. 2. The estimated marker effects must be used in subsequent
selection cycles to obtain GEBVs that act as predictors of the
individual breeding values in the testing population (where there is
only marker information). 3. The GEBV values must be composed entirely
of the additive genetic effects. 4. Phenotypes must be used to estimate
all marker effects in the training population, and <em>not</em> to make
selections in the testing population (Heffner et al., 2009; Lorenz et
al., 2011).</p>
</div>
<div class="section level3">
<h3 id="the-lgsi-and-its-parameters">The LGSI and Its Parameters<a class="anchor" aria-label="anchor" href="#the-lgsi-and-its-parameters"></a>
</h3>
<p>The objective of LGSI is to predict the net genetic merit
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>H</mi><mo>=</mo><mi>ùê∞</mi><mi>‚Ä≤</mi><mi>ùê†</mi></mrow><annotation encoding="application/x-tex">H = \mathbf{w}'\mathbf{g}</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùê†</mi><annotation encoding="application/x-tex">\mathbf{g}</annotation></semantics></math>
is a vector of unobservable true breeding values, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùê∞</mi><annotation encoding="application/x-tex">\mathbf{w}</annotation></semantics></math>
is a vector of economic weights. Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùõÑ</mi><annotation encoding="application/x-tex">\mathbf{\gamma}</annotation></semantics></math>
be the vector of known Genomic Estimated Breeding Values (GEBVs). The
basic LGSI equation is:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>I</mi><mi mathvariant="normal">G</mi></msub><mo>=</mo><msup><mi>ùõÉ</mi><mi>‚Ä≤</mi></msup><mi>ùõÑ</mi></mrow><annotation encoding="application/x-tex"> {I}_{\mathrm{G}}={\boldsymbol{\beta}}^{\prime}\boldsymbol{\gamma} </annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùõÉ</mi><annotation encoding="application/x-tex">\boldsymbol{\beta}</annotation></semantics></math>
is the unknown variable vector of optimal weights. For LGSI, it has been
shown that maximizing the accuracy of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow></msub><annotation encoding="application/x-tex">\rho_{HI_G}</annotation></semantics></math>
results simply in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ùõÉ</mi><mo>=</mo><mi>ùê∞</mi></mrow><annotation encoding="application/x-tex">\boldsymbol{\beta} = \mathbf{w}</annotation></semantics></math>.</p>
<p>The main advantage of the LGSI over the conventional Linear
Phenotypic Selection Index (LPSI) lies in the possibility of reducing
the intervals between selection cycles from roughly 4 years to 1.5 years
in plant breeding. The selection response equation adjusts for cycle
length
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>L</mi><mi>G</mi></msub><annotation encoding="application/x-tex">L_G</annotation></semantics></math>:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><msub><mi>I</mi><mi>G</mi></msub></msub><mo>=</mo><mfrac><msub><mi>k</mi><mi>I</mi></msub><msub><mi>L</mi><mi>G</mi></msub></mfrac><msqrt><mrow><msup><mi>ùê∞</mi><mi>‚Ä≤</mi></msup><mi>ùö™</mi><mi>ùê∞</mi></mrow></msqrt></mrow><annotation encoding="application/x-tex"> {R}_{I_G}=\frac{k_I}{L_G}\sqrt{{\mathbf{w}}^{\prime}\boldsymbol{\Gamma} \mathbf{w}} </annotation></semantics></math>
Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùö™</mi><annotation encoding="application/x-tex">\boldsymbol{\Gamma}</annotation></semantics></math>
is the covariance matrix of the GEBVs.</p>
</div>
<div class="section level3">
<h3 id="statistical-lgsi-properties">Statistical LGSI Properties<a class="anchor" aria-label="anchor" href="#statistical-lgsi-properties"></a>
</h3>
<p>The LGSI operates as a direct genomic extension of the LPSI theory,
retaining the exact same statistical properties: 1. The variance of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>G</mi></msub><annotation encoding="application/x-tex">I_G</annotation></semantics></math>
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>œÉ</mi><msub><mi>I</mi><mi>G</mi></msub><mn>2</mn></msubsup><annotation encoding="application/x-tex">\sigma_{I_G}^2</annotation></semantics></math>)
and the covariance between
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>G</mi></msub><annotation encoding="application/x-tex">I_G</annotation></semantics></math>
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>œÉ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow></msub><annotation encoding="application/x-tex">\sigma_{HI_G}</annotation></semantics></math>)
are equal
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>œÉ</mi><msub><mi>I</mi><mi>G</mi></msub><mn>2</mn></msubsup><mo>=</mo><msub><mi>œÉ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow></msub></mrow><annotation encoding="application/x-tex">\sigma_{I_G}^2 = \sigma_{HI_G}</annotation></semantics></math>).
2. The maximized correlation between
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>G</mi></msub><annotation encoding="application/x-tex">I_G</annotation></semantics></math>
(the LGSI accuracy) is equal to
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow></msub><mo>=</mo><mfrac><msub><mi>œÉ</mi><msub><mi>I</mi><mi>G</mi></msub></msub><msub><mi>œÉ</mi><mi>H</mi></msub></mfrac></mrow><annotation encoding="application/x-tex">\rho_{HI_G} = \frac{\sigma_{I_G}}{\sigma_H}</annotation></semantics></math>.
3. The variance of the predicted error,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>V</mi><mi>a</mi><mi>r</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>H</mi><mo>‚àí</mo><msub><mi>I</mi><mi>G</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>‚àí</mo><msubsup><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><msubsup><mi>œÉ</mi><mi>H</mi><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">Var(H - I_G) = (1 - \rho_{HI_G}^2)\sigma_H^2</annotation></semantics></math>,
is minimal. 4. The total variance of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
explained by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>G</mi></msub><annotation encoding="application/x-tex">I_G</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>œÉ</mi><msub><mi>I</mi><mi>G</mi></msub><mn>2</mn></msubsup><mo>=</mo><msubsup><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow><mn>2</mn></msubsup><msubsup><mi>œÉ</mi><mi>H</mi><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">\sigma_{I_G}^2 = \rho_{HI_G}^2 \sigma_H^2</annotation></semantics></math>.</p>
</div>
<div class="section level3">
<h3 id="lgsi-vs-lpsi-efficiency">LGSI vs LPSI Efficiency<a class="anchor" aria-label="anchor" href="#lgsi-vs-lpsi-efficiency"></a>
</h3>
<p>To formalize the efficiency between the Genomic (LGSI) and Phenotypic
(LPSI) selection index efficiency, we use the parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>Œª</mi><mi>G</mi></msub><annotation encoding="application/x-tex">\lambda_G</annotation></semantics></math>:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mi>G</mi></msub><mo>=</mo><mn>100</mn><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>Œª</mi><mi>G</mi></msub><mo>‚àí</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex"> p_G=100\left(\lambda_G-1\right) </annotation></semantics></math></p>
<p>If
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mi>G</mi></msub><mo>&gt;</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">p_G &gt; 0</annotation></semantics></math>,
LGSI efficiency is greater than LPSI efficiency; if
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mi>G</mi></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">p_G = 0</annotation></semantics></math>,
both are equal, and if
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mi>G</mi></msub><mo>&lt;</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">p_G &lt; 0</annotation></semantics></math>,
LPSI is more efficient. Additionally, accounting for the interval
lengths between selection cycles, LGSI will be strictly more efficient
than LPSI when:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>L</mi><mi>G</mi></msub><mo>&lt;</mo><mfrac><msub><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>G</mi></msub></mrow></msub><msub><mi>h</mi><mi>I</mi></msub></mfrac><msub><mi>L</mi><mi>P</mi></msub></mrow><annotation encoding="application/x-tex"> L_G &lt; \frac{\rho_{HI_G}}{h_I}L_P </annotation></semantics></math></p>
</div>
<div class="section level3">
<h3 id="genomic-estimated-breeding-values-gebvs">Genomic Estimated Breeding Values (GEBVs)<a class="anchor" aria-label="anchor" href="#genomic-estimated-breeding-values-gebvs"></a>
</h3>
<p>Because the <code>selection.index</code> package does not perform
prediction modeling itself, the GEBVs must be calculated prior. In this
example, we generate GEBVs
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùõÑ</mi><annotation encoding="application/x-tex">\boldsymbol{\gamma}</annotation></semantics></math>)
using an unpenalized Ridge Regression, where markers estimate trait
values. For a rigorous methodology it is acceptable to generate GEBVs
using Bayesian Alphabets, rrBLUP, or equivalent prediction models.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zankrut20/selection.index" class="external-link">selection.index</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.stats.ox.ac.uk/pub/MASS4/" class="external-link">MASS</a></span><span class="op">)</span> <span class="co"># For robust ridge regression implementation (lm.ridge)</span></span>
<span></span>
<span><span class="co"># 1. Load the built-in synthetic maize datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_pheno"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_geno"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure markers are numeric matrices</span></span>
<span><span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">maize_geno</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure pheno lines up perfectly with geno</span></span>
<span><span class="co"># (Note: maize_pheno contains 4 repetitions per genotype; we take the means)</span></span>
<span><span class="va">Y_agg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mean_performance.html">mean_performance</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Yield"</span>, <span class="st">"PlantHeight"</span>, <span class="st">"DaysToMaturity"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  genotypes <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span>,</span>
<span>  replications <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Block</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(2 * EMS/r): NaNs produced</span></span>
<span><span class="co">#&gt; Warning in sqrt(EMS): NaNs produced</span></span>
<span></span>
<span><span class="co"># Sort both datasets to ensure identical ordering</span></span>
<span><span class="va">Y_agg</span> <span class="op">&lt;-</span> <span class="va">Y_agg</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Y_agg</span><span class="op">$</span><span class="va">Genotypes</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">X</span> <span class="op">&lt;-</span> <span class="va">X</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Ensure overlapping genotypes</span></span>
<span><span class="va">common_genotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">Y_agg</span><span class="op">$</span><span class="va">Genotypes</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Y_agg</span> <span class="op">&lt;-</span> <span class="va">Y_agg</span><span class="op">[</span><span class="va">Y_agg</span><span class="op">$</span><span class="va">Genotypes</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">common_genotypes</span>, <span class="op">]</span></span>
<span><span class="va">X</span> <span class="op">&lt;-</span> <span class="va">X</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">common_genotypes</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Select only multi-traits relevant to breeding</span></span>
<span><span class="co"># Yield, PlantHeight, DaysToMaturity</span></span>
<span><span class="va">Y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">Y_agg</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Yield"</span>, <span class="st">"PlantHeight"</span>, <span class="st">"DaysToMaturity"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>We now calculate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùõÑ</mi><annotation encoding="application/x-tex">\boldsymbol{\gamma}</annotation></semantics></math>
(the <code>gebv_mat</code>):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Simulate Genomic Estimated Breeding Values (GEBVs) using Ridge Regression</span></span>
<span><span class="co"># In best practice, you'd use cross-validation or a separate training/testing set</span></span>
<span><span class="co"># We use lambda = 100 to handle the p &gt;&gt; n dimensionality problem</span></span>
<span><span class="va">gebv_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">gebv_mat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Fit a ridge regression model for trait `i` using markers `X`</span></span>
<span>  <span class="va">model_ridge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MASS/man/lm.ridge.html" class="external-link">lm.ridge</a></span><span class="op">(</span><span class="va">Y</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span> <span class="op">~</span> <span class="va">X</span>, lambda <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Predict values using coef() which correctly un-scales the coefficients: Include intercept</span></span>
<span>  <span class="va">betas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">model_ridge</span><span class="op">)</span></span>
<span>  <span class="va">intercept</span> <span class="op">&lt;-</span> <span class="va">betas</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">betas</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="co"># Exclude intercept</span></span>
<span></span>
<span>  <span class="co"># Calculate the GEBV for the trait</span></span>
<span>  <span class="va">gebv_mat</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">intercept</span> <span class="op">+</span> <span class="op">(</span><span class="va">X</span> <span class="op"><a href="https://rdrr.io/r/base/matmult.html" class="external-link">%*%</a></span> <span class="va">beta</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gebv_mat</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt;         Yield PlantHeight DaysToMaturity</span></span>
<span><span class="co">#&gt; [1,] 8241.151    219.4007       115.8738</span></span>
<span><span class="co">#&gt; [2,] 7965.285    208.8754       116.4623</span></span>
<span><span class="co">#&gt; [3,] 7832.597    204.5494       116.4592</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="implementing-lgsi">Implementing LGSI<a class="anchor" aria-label="anchor" href="#implementing-lgsi"></a>
</h3>
<p>With <code>gebv_mat</code> in hand, it is simple to calculate the
LGSI metrics using <code><a href="../reference/lgsi.html">lgsi()</a></code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 2. Compute Trait Covariances</span></span>
<span><span class="va">pmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cov</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span> <span class="co"># Phenotypic Covariance (Approximation)</span></span>
<span></span>
<span><span class="co"># In optimal practice, the true Genomic Covariance Matrix (\Gamma) is</span></span>
<span><span class="co"># estimated using Restricted Maximum Likelihood (REML) utilizing both general</span></span>
<span><span class="co"># phenotypic and genotypic information. Here, we simulate \Gamma as proportional</span></span>
<span><span class="co"># to the phenotypic variance (assuming a high heritability correlation).</span></span>
<span><span class="va">gmat</span> <span class="op">&lt;-</span> <span class="va">pmat</span> <span class="op">*</span> <span class="fl">0.4</span> <span class="co"># Approximate Genotypic Covariance (Approximating \Gamma)</span></span>
<span></span>
<span><span class="co"># 3. Define Economic Weights</span></span>
<span><span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  Trait <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Yield"</span>, <span class="st">"PlantHeight"</span>, <span class="st">"DaysToMaturity"</span><span class="op">)</span>,</span>
<span>  Weight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="op">-</span><span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">wmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_mat.html">weight_mat</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 4. Calculate Linear Genomic Selection Index (LGSI)</span></span>
<span><span class="co"># For the testing population where we only use genomic values</span></span>
<span><span class="va">lgsi_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lgsi.html">lgsi</a></span><span class="op">(</span></span>
<span>  gebv_mat <span class="op">=</span> <span class="va">gebv_mat</span>,</span>
<span>  gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">wmat</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Output Summary</span></span>
<span><span class="va">lgsi_result</span><span class="op">$</span><span class="va">summary</span></span>
<span><span class="co">#&gt;   b.1  b.2  b.3       GA      PRE hI2 rHI</span></span>
<span><span class="co">#&gt; 1   5 -0.1 -0.1 1718.838 171883.8   1   1</span></span></code></pre></div>
<p>The LGSI function output shows the calculated standard deviation
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>œÉ</mi><mi>I</mi></msub><annotation encoding="application/x-tex">\sigma_I</annotation></semantics></math>),
index heritability
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>),
the overall net genetic advance
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>G</mi><mi>A</mi></mrow><annotation encoding="application/x-tex">GA</annotation></semantics></math>),
and expected gains per trait
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Œî</mi><mi>H</mi></mrow><annotation encoding="application/x-tex">\Delta H</annotation></semantics></math>)
associated with genomic selection.</p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="the-combined-linear-genomic-selection-index">5.2 The Combined Linear Genomic Selection Index<a class="anchor" aria-label="anchor" href="#the-combined-linear-genomic-selection-index"></a>
</h2>
<p>The Combined LGSI (CLGSI) developed by Dekkers (2007) is a slightly
modified version of the LMSI, which, instead of using the raw marker
scores, uses the GEBVs and the phenotypic information jointly to predict
the net genetic merit.</p>
<p>The CLGSI model represents:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>I</mi><mi>C</mi></msub><mo>=</mo><msubsup><mi>ùõÉ</mi><mi>ùê≤</mi><mi>‚Ä≤</mi></msubsup><mi>ùê≤</mi><mo>+</mo><msubsup><mi>ùõÉ</mi><mi>G</mi><mi>‚Ä≤</mi></msubsup><mi>ùõÑ</mi></mrow><annotation encoding="application/x-tex"> {I}_C={\boldsymbol{\beta}}_{\mathbf{y}}^{\prime}\mathbf{y}+{\boldsymbol{\beta}}_G^{\prime}\boldsymbol{\gamma} </annotation></semantics></math></p>
<p>Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùê≤</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>
is the predicted phenotype and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ùõÑ</mi><annotation encoding="application/x-tex">\boldsymbol{\gamma}</annotation></semantics></math>
is the subset of genetic markers. As
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>‚Üí</mo><mi>‚àû</mi></mrow><annotation encoding="application/x-tex">p \rightarrow \infty</annotation></semantics></math>,
the true variance approaches the genomic predicted variance, and CLGSI
becomes identical to LGSI. Because the CLGSI uses GEBVs and phenotypic
information jointly, it can ideally complement prediction workflows
inside training populations where field trials are simultaneously
held.</p>
<div class="section level3">
<h3 id="statistical-properties-of-the-clgsi">Statistical Properties of the CLGSI<a class="anchor" aria-label="anchor" href="#statistical-properties-of-the-clgsi"></a>
</h3>
<p>Similar to LGSI, the CLGSI holds strict conditions ensuring
theoretical validation constraints: 1.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>œÉ</mi><msub><mi>I</mi><mi>C</mi></msub><mn>2</mn></msubsup><mo>=</mo><msub><mi>œÉ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>C</mi></msub></mrow></msub></mrow><annotation encoding="application/x-tex">\sigma_{I_C}^2 = \sigma_{HI_C}</annotation></semantics></math>,
meaning the variance of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>C</mi></msub><annotation encoding="application/x-tex">I_C</annotation></semantics></math>
and the covariance between
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>C</mi></msub><annotation encoding="application/x-tex">I_C</annotation></semantics></math>
are unified. 2. The maximized correlation between
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>C</mi></msub><annotation encoding="application/x-tex">I_C</annotation></semantics></math>
operates strictly as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>C</mi></msub></mrow></msub><mo>=</mo><mfrac><msub><mi>œÉ</mi><msub><mi>I</mi><mi>C</mi></msub></msub><msub><mi>œÉ</mi><mi>H</mi></msub></mfrac></mrow><annotation encoding="application/x-tex">\rho_{HI_C} = \frac{\sigma_{I_C}}{\sigma_H}</annotation></semantics></math>.
3. The variance of the predicted error,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>V</mi><mi>a</mi><mi>r</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>H</mi><mo>‚àí</mo><msub><mi>I</mi><mi>C</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>‚àí</mo><msubsup><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>C</mi></msub></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><msubsup><mi>œÉ</mi><mi>H</mi><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">Var(H - I_C) = (1 - \rho_{HI_C}^2)\sigma_H^2</annotation></semantics></math>,
is mathematically minimal. 4. The total variance of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
explained by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>C</mi></msub><annotation encoding="application/x-tex">I_C</annotation></semantics></math>
scales linearly via
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>œÉ</mi><msub><mi>I</mi><mi>C</mi></msub><mn>2</mn></msubsup><mo>=</mo><msubsup><mi>œÅ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>C</mi></msub></mrow><mn>2</mn></msubsup><msubsup><mi>œÉ</mi><mi>H</mi><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">\sigma_{I_C}^2 = \rho_{HI_C}^2 \sigma_H^2</annotation></semantics></math>.</p>
</div>
<div class="section level3">
<h3 id="implementing-clgsi">Implementing CLGSI<a class="anchor" aria-label="anchor" href="#implementing-clgsi"></a>
</h3>
<p>You can measure the CLGSI utilizing the <code><a href="../reference/clgsi.html">clgsi()</a></code> function
and feeding it both the observed phenotypes and estimated genomic
variants.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clgsi_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clgsi.html">clgsi</a></span><span class="op">(</span></span>
<span>  phen_mat <span class="op">=</span> <span class="va">Y</span>, <span class="co"># Observed phenotypic data</span></span>
<span>  gebv_mat <span class="op">=</span> <span class="va">gebv_mat</span>, <span class="co"># Genomic Estimated Breeding Values</span></span>
<span>  pmat <span class="op">=</span> <span class="va">pmat</span>, <span class="co"># Expected Phenotypic traits covariance</span></span>
<span>  gmat <span class="op">=</span> <span class="va">gmat</span>, <span class="co"># Expected Genotypic traits covariance</span></span>
<span>  wmat <span class="op">=</span> <span class="va">wmat</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">clgsi_result</span><span class="op">$</span><span class="va">summary</span></span>
<span><span class="co">#&gt;       b_y.1    b_y.2    b_y.3    b_g.1    b_g.2    b_g.3       GA      PRE hI2</span></span>
<span><span class="co">#&gt; 1 -241.2056 -69.0278 -11711.7 301.3417 -12.6428 13421.05 9325.477 932547.7   1</span></span>
<span><span class="co">#&gt;   rHI</span></span>
<span><span class="co">#&gt; 1   1</span></span></code></pre></div>
<p>Notice that the Combined LGSI has two sets of optimized unitless
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>Œ≤</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>
weightings‚Äîone for the phenotype values (<code>b_y</code>), and one for
the overall marker value (<code>b_g</code>) per trait. Selection
Response
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>R</mi><annotation encoding="application/x-tex">R</annotation></semantics></math>)
represents the final expected genetic aggregate improvement.</p>
</div>
</div>
<div class="section level2">
<h2 id="limitations-and-caveats">Limitations and Caveats<a class="anchor" aria-label="anchor" href="#limitations-and-caveats"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Dimensionality Problem</strong>: Often
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>‚â´</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">p \gg n</annotation></semantics></math>,
requiring regularization techniques like Ridge Regression (as
demonstrated in <code>lm.ridge</code>) to derive the Genomic Estimated
Breeding Values without singular/non-invertible covariance
anomalies.</li>
<li>
<strong>Cycle Length</strong>: The genomic methods (LGSI, CLGSI) may
occasionally show a lower index accuracy unadjusted than the phenotypic
methods initially, but typically provide vastly more improvement per
unit of time
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>L</mi><mi>G</mi></msub><annotation encoding="application/x-tex">L_G</annotation></semantics></math>)
because of how heavily compressed cycle iterations are when phenotyping
is skipped.</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zankrut Goyani.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
