<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Linear Marker and Genome-Wide Selection Indices â€¢ selection.index</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><script src="../extra.js"></script><meta property="og:title" content="Linear Marker and Genome-Wide Selection Indices">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">selection.index</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started with selection.index</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-selection-index.html">The Linear Phenotypic Selection Index Theory</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-phenotypic-selection-indices.html">Constrained Linear Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-marker-selection-indices.html">Linear Marker and Genome-Wide Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-selection-indices.html">Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-genomic-selection-indices.html">Constrained Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-eigen-selection-index-methods.html">Linear Phenotypic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-eigen-selection-index-methods.html">Linear Genomic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/multistage-linear-selection-indices.html">Multistage Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/stochastic-simulation-linear-phenotypic-indices.html">Stochastic Simulation of Linear Phenotypic Selection Indices</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zankrut20/selection.index/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Linear Marker and Genome-Wide Selection Indices</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zankrut20/selection.index/blob/master/vignettes/linear-marker-selection-indices.Rmd" class="external-link"><code>vignettes/linear-marker-selection-indices.Rmd</code></a></small>
      <div class="d-none name"><code>linear-marker-selection-indices.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>There are two main linear selection indices employed in
marker-assisted selection (MAS) to predict the net genetic merit and to
select individual candidates as parents for the next generation: the
<strong>linear marker selection index (LMSI)</strong> and the
<strong>genome-wide LMSI (GW-LMSI)</strong>.</p>
<p>Both indices maximize the expected genetic gain per trait by
combining molecular and phenotypic data. The LMSI requires a prior step
of selecting markers significantly linked to the quantitative trait loci
(QTL) to calculate a â€œmarker scoreâ€. The GW-LMSI is a single-stage
procedure that treats information at each individual marker as a
separate trait, entering all markers and phenotypic information into a
single index.</p>
<p>In this vignette, we demonstrate how to apply these indices using the
<code>selection.index</code> package on phenotypic
(<code>maize_pheno</code>) and genotypic (<code>maize_geno</code>)
data.</p>
<div class="section level3">
<h3 id="sample-data-preparation">Sample Data Preparation<a class="anchor" aria-label="anchor" href="#sample-data-preparation"></a>
</h3>
<p>We use the synthetic maize datasets built into the package to
calculate the phenotypic, genotypic, marker, and covariance matrices
necessary for molecular index modeling.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zankrut20/selection.index" class="external-link">selection.index</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the built-in maize phenotype and genotype datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_pheno"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_geno"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. Prepare Phenotypic Data</span></span>
<span><span class="co"># We select three traits for our demonstration</span></span>
<span><span class="va">traits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Yield"</span>, <span class="st">"PlantHeight"</span>, <span class="st">"DaysToMaturity"</span><span class="op">)</span></span>
<span><span class="va">phen_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate Genotypic (gmat) and Phenotypic (pmat) covariance matrices</span></span>
<span><span class="va">gmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gen_varcov.html">gen_varcov</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Block</span><span class="op">)</span></span>
<span><span class="va">pmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phen_varcov.html">phen_varcov</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Block</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate phenotypic data by Genotype to match marker data dimensions</span></span>
<span><span class="va">agg_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Genotype <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span><span class="op">)</span>, FUN <span class="op">=</span> <span class="va">mean</span><span class="op">)</span></span>
<span><span class="va">phen_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">agg_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Prepare Genotypic Data</span></span>
<span><span class="co"># Extract only the marker columns (dropping the Genotype ID column)</span></span>
<span><span class="va">marker_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">maize_geno</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. Define Economic Weights</span></span>
<span><span class="co"># Suppose we want to improve Yield strongly, decrease Plant Height, and lightly increase Days to Maturity</span></span>
<span><span class="va">wmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_mat.html">weight_mat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Trait <span class="op">=</span> <span class="va">traits</span>, Weight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="the-linear-marker-selection-index-lmsi">1. The Linear Marker Selection Index (LMSI)<a class="anchor" aria-label="anchor" href="#the-linear-marker-selection-index-lmsi"></a>
</h2>
<p>The Linear Marker Selection Index (LMSI) requires an explicit prior
step: calculating the <strong>marker score</strong>
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ¬</mi><annotation encoding="application/x-tex">\mathbf{s}</annotation></semantics></math>).</p>
<p>Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ²</mi><mi>i</mi></msub><mo>=</mo><msub><mi>ğ </mi><mi>i</mi></msub><mo>+</mo><msub><mi>ğ</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf{y}_i = \mathbf{g}_i + \mathbf{e}_i</annotation></semantics></math>
be the vector of phenotypic traits. The unobservable genetic value
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>g</mi><mi>i</mi></msub><annotation encoding="application/x-tex">g_i</annotation></semantics></math>
can be predicted using a linear combination of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
selected markers structurally linked to the QTL that affect the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>th
trait:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ¬</mi><mi>i</mi></msub><mo>=</mo><munderover><mo>âˆ‘</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></munderover><msub><mi>ğ›‰</mi><mi>j</mi></msub><msub><mi>ğ±</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex"> \mathbf{s}_i=\sum \limits_{j=1}^M \boldsymbol{\theta}_j \mathbf{x}_j </annotation></semantics></math></p>
<p>Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ¬</mi><mi>i</mi></msub><annotation encoding="application/x-tex">\mathbf{s}_i</annotation></semantics></math>
is the marker score,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ›‰</mi><mi>j</mi></msub><annotation encoding="application/x-tex">\boldsymbol{\theta}_j</annotation></semantics></math>
is the regression coefficient, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ±</mi><mi>j</mi></msub><annotation encoding="application/x-tex">\mathbf{x}_j</annotation></semantics></math>
is the coded value of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>th
markers.</p>
<p>The LMSI combines the phenotypic scores
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ²</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>)
and marker scores
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ¬</mi><annotation encoding="application/x-tex">\mathbf{s}</annotation></semantics></math>)
to predict the net genetic merit
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>:</p>
<p><span class="math display">$$
I_M={\boldsymbol{\beta}}_y^{\prime}\mathbf{y}+{\boldsymbol{\beta}}_s^{\prime}\mathbf{s}=\left[{\boldsymbol{\beta}}_y^{\prime}\kern0.5em
{\boldsymbol{\beta}}_s^{\prime}\right]\left[\begin{array}{c}\mathbf{y}\\
{}\mathbf{s}\end{array}\right] $$</span></p>
<div class="section level3">
<h3 id="practical-application-lmsi">Practical Application: LMSI<a class="anchor" aria-label="anchor" href="#practical-application-lmsi"></a>
</h3>
<p>To apply the LMSI, we must first estimate the marker scores for each
trait. A simple way is to perform multiple linear regression of the
phenotypic values on the marker data, selecting only significant
markers, and using those coefficients to calculate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ¬</mi><annotation encoding="application/x-tex">\mathbf{s}</annotation></semantics></math>.</p>
<p>For simplicity in this tutorial, we will use all markers in a linear
model (ridge regression) to predict the traits.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate marker scores via a simple Ridge Regression</span></span>
<span><span class="co"># (We use MASS::lm.ridge or glmnet in practice, but for the vignette we approximate</span></span>
<span><span class="co"># the prediction using the marker matrix directly)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.stats.ox.ac.uk/pub/MASS4/" class="external-link">MASS</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">marker_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">phen_mat</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">phen_mat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">marker_scores</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">phen_mat</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">phen_mat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Fit ridge regression to handle multicollinearity among markers</span></span>
<span>  <span class="co"># Note: A lambda must be chosen carefully in real scenarios</span></span>
<span>  <span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MASS/man/lm.ridge.html" class="external-link">lm.ridge</a></span><span class="op">(</span><span class="va">phen_mat</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span> <span class="op">~</span> <span class="va">marker_mat</span>, lambda <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>  <span class="co"># Calculate predicted marker scores</span></span>
<span>  <span class="va">marker_scores</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">marker_mat</span>, center <span class="op">=</span> <span class="va">fit</span><span class="op">$</span><span class="va">xm</span>, scale <span class="op">=</span> <span class="va">fit</span><span class="op">$</span><span class="va">scales</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/matmult.html" class="external-link">%*%</a></span> <span class="va">fit</span><span class="op">$</span><span class="va">coef</span> <span class="op">+</span> <span class="va">fit</span><span class="op">$</span><span class="va">ym</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Now that we have
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ©</mi><mi>ğ¦</mi><mi>ğš</mi><mi>ğ­</mi></mrow><annotation encoding="application/x-tex">\mathbf{pmat}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ </mi><mi>ğ¦</mi><mi>ğš</mi><mi>ğ­</mi></mrow><annotation encoding="application/x-tex">\mathbf{gmat}</annotation></semantics></math>,
the phenotypic matrix
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ²</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>,
and the estimated marker score matrix
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ¬</mi><annotation encoding="application/x-tex">\mathbf{s}</annotation></semantics></math>,
we can calculate the <strong>Linear Marker Selection Index</strong>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the LMSI</span></span>
<span><span class="va">lmsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lmsi.html">lmsi</a></span><span class="op">(</span></span>
<span>  phen_mat <span class="op">=</span> <span class="va">phen_mat</span>,</span>
<span>  marker_scores <span class="op">=</span> <span class="va">marker_scores</span>,</span>
<span>  pmat <span class="op">=</span> <span class="va">pmat</span>,</span>
<span>  gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">wmat</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the LMSI coefficient summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">lmsi_res</span><span class="op">$</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="co">#&gt;            Trait   Component            b    w   Delta_H</span></span>
<span><span class="co">#&gt; 1          Yield   Phenotype     521.4950  2.0 731752.99</span></span>
<span><span class="co">#&gt; 2    PlantHeight   Phenotype    2085.7769 -1.0  19178.95</span></span>
<span><span class="co">#&gt; 3 DaysToMaturity   Phenotype  223294.4615  0.5  10659.56</span></span>
<span><span class="co">#&gt; 4          Yield MarkerScore    -532.8818  2.0 731752.99</span></span>
<span><span class="co">#&gt; 5    PlantHeight MarkerScore   -2003.1664 -1.0  19178.95</span></span>
<span><span class="co">#&gt; 6 DaysToMaturity MarkerScore -229077.9808  0.5  10659.56</span></span>
<span></span>
<span><span class="co"># Display the expected genetic gains per trait</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">lmsi_res</span><span class="op">$</span><span class="va">Delta_G</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>The output yields the relative contribution weights of the phenotypes
(<code>b_y.*</code>) versus the markers (<code>b_s.*</code>), and
calculates the overall Expected Genetic Advance.</p>
</div>
</div>
<div class="section level2">
<h2 id="the-genome-wide-linear-marker-selection-index-gw-lmsi">2. The Genome-Wide Linear Marker Selection Index (GW-LMSI)<a class="anchor" aria-label="anchor" href="#the-genome-wide-linear-marker-selection-index-gw-lmsi"></a>
</h2>
<p>The genome-wide linear selection index (GW-LMSI) bypasses the
two-stage regression scoring step of the LMSI. Instead, it treats
information at each individual marker as a separate trait. All
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
marker information enters together with phenotypic information into a
single index.</p>
<p>The GW-LMSI combines the trait phenotypic values
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ²</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>)
and the molecular information
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ¦</mi><annotation encoding="application/x-tex">\mathbf{m}</annotation></semantics></math>)
to predict
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>:</p>
<p><span class="math display">$$
I_W={\boldsymbol{\beta}}_y^{\prime}\mathbf{y}+{\boldsymbol{\beta}}_m^{\prime}\mathbf{m}=\left[{\boldsymbol{\beta}}_y^{\prime}\kern0.5em
{\boldsymbol{\beta}}_m^{\prime}\right]\left[\begin{array}{c}\mathbf{y}\\
{}\mathbf{m}\end{array}\right] $$</span></p>
<p>Because the number of markers
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
often wildly exceeds the number of individuals
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
(the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>â‰«</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">p \gg n</annotation></semantics></math>
problem), the covariance matrix of the markers
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğŒ</mi><annotation encoding="application/x-tex">\mathbf{M}</annotation></semantics></math>)
and the covariance between phenotypes and markers
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ–</mi><annotation encoding="application/x-tex">\mathbf{W}</annotation></semantics></math>)
are generally singular.</p>
<p>Thus, calculating the index coefficients
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ›ƒ</mi><mi>y</mi></msub><annotation encoding="application/x-tex">{\boldsymbol{\beta}}_y</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ›ƒ</mi><mi>m</mi></msub><annotation encoding="application/x-tex">{\boldsymbol{\beta}}_m</annotation></semantics></math>
typically requires generalized inverses or regularization (e.g.Â Ridge
Regression / Tikhonov regularization applied to the covariance
matrices).</p>
<div class="section level3">
<h3 id="practical-application-gw-lmsi">Practical Application: GW-LMSI<a class="anchor" aria-label="anchor" href="#practical-application-gw-lmsi"></a>
</h3>
<p>The <code>selection.index</code> package manages the complex
construction of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğŒ</mi><annotation encoding="application/x-tex">\mathbf{M}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ–</mi><annotation encoding="application/x-tex">\mathbf{W}</annotation></semantics></math>
covariance matrices asymptotically and handles singularity issues using
Tikhonov regularization (<code>lambda</code>).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We use all 500 markers in the maize_geno matrix</span></span>
<span><span class="co"># Since p (500) approaches n (600), we supply a ridge regularization lambda</span></span>
<span><span class="co"># to ensure the covariance matrices are invertible.</span></span>
<span><span class="va">gw_lmsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gw_lmsi.html">gw_lmsi</a></span><span class="op">(</span></span>
<span>  marker_mat <span class="op">=</span> <span class="va">marker_mat</span>,</span>
<span>  trait_mat <span class="op">=</span> <span class="va">phen_mat</span>,</span>
<span>  gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">wmat</span>,</span>
<span>  lambda <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the phenotypic weighting coefficients</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gw_lmsi_res</span><span class="op">$</span><span class="va">b_y</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span></span>
<span><span class="co"># We can also check the condition number to observe the matrix stability</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gw_lmsi_res</span><span class="op">$</span><span class="va">condition_number</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 6.629131</span></span>
<span></span>
<span><span class="co"># And summarize the overall index statistics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gw_lmsi_res</span><span class="op">$</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="co">#&gt;      Metric    Value</span></span>
<span><span class="co">#&gt; 1       rHI   0.0252</span></span>
<span><span class="co">#&gt; 2       hI2   1.0000</span></span>
<span><span class="co">#&gt; 3   sigma_I 397.3752</span></span>
<span><span class="co">#&gt; 4         R 819.7850</span></span>
<span><span class="co">#&gt; 5        GA 840.7368</span></span>
<span><span class="co">#&gt; 6       PRE       NA</span></span>
<span><span class="co">#&gt; 7 n_markers 499.0000</span></span>
<span><span class="co">#&gt; 8    lambda   0.0500</span></span></code></pre></div>
<p>The GW-LMSI automatically models all the molecular variance in one
step. While it requires computationally heavier matrix inversion (often
mitigated by regularization), it tends to predict the net genetic merit
more efficiently than standard LMSI when trait heritability is very low
or when capturing many small QTL regressions requires immense sample
sizes.</p>
</div>
</div>
<div class="section level2">
<h2 id="advanced-concepts-and-limitations">3. Advanced Concepts and Limitations<a class="anchor" aria-label="anchor" href="#advanced-concepts-and-limitations"></a>
</h2>
<div class="section level3">
<h3 id="basic-conditions-for-constructing-the-lmsi">3.1 Basic Conditions for Constructing the LMSI<a class="anchor" aria-label="anchor" href="#basic-conditions-for-constructing-the-lmsi"></a>
</h3>
<p>According to Lande and Thompson (1990), applying the LMSI correctly
requires several primary statistical and biological conditions to be met
in the breeding population:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Linkage Disequilibrium:</strong> The markers and the
quantitative trait loci (QTL) must be in linkage disequilibrium
(LD).</li>
<li>
<strong>Additivity:</strong> QTL effects should be combined
additively both within and between loci (no epistasis).</li>
<li>
<strong>Coupling Mode:</strong> The QTL should be in coupling mode
where parental lines hold uniformly positive or negative alleles.</li>
<li>
<strong>Genetic Architecture:</strong> Traits should be affected by
a few QTLs with large effects rather than exclusively by many
small-effect QTLs.</li>
<li>
<strong>Heritability:</strong> The traits must inherently have a low
heritability
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>).
If heritabilities are high, phenotypic selection alone is immensely
effective.</li>
<li>
<strong>Marker Identification:</strong> Markers correlated with the
traits of interest must first be successfully identified.</li>
</ol>
</div>
<div class="section level3">
<h3 id="the-lmsi-paradox-and-efficiency-comparisons">3.2 The â€œLMSI Paradoxâ€ and Efficiency Comparisons<a class="anchor" aria-label="anchor" href="#the-lmsi-paradox-and-efficiency-comparisons"></a>
</h3>
<p>Will a linear marker selection index always dramatically outperform
standard phenotypic selection? Not necessarily.</p>
<p>Lande and Thompson (1990) showed that LMSI efficiency relative to
basic phenotypic selection
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>Î»</mi><mi>M</mi></msub><annotation encoding="application/x-tex">\lambda_M</annotation></semantics></math>)
for a single trait is:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Î»</mi><mi>M</mi></msub><mo>=</mo><mfrac><msub><mi>R</mi><mi>M</mi></msub><mi>R</mi></mfrac><mo>=</mo><msqrt><mrow><mfrac><mi>q</mi><msup><mi>h</mi><mn>2</mn></msup></mfrac><mo>+</mo><mfrac><msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>âˆ’</mo><mi>q</mi><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mrow><mn>1</mn><mo>âˆ’</mo><msup><mrow><mi>q</mi><mi>h</mi></mrow><mn>2</mn></msup></mrow></mfrac></mrow></msqrt></mrow><annotation encoding="application/x-tex"> \lambda_M=\frac{R_M}{R}=\sqrt{\frac{q}{h^2}+\frac{{\left(1-q\right)}^2}{1-{qh}^2}} </annotation></semantics></math></p>
<p>Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mfrac><msubsup><mi>Ïƒ</mi><mi>s</mi><mn>2</mn></msubsup><msubsup><mi>Ïƒ</mi><mi>g</mi><mn>2</mn></msubsup></mfrac></mrow><annotation encoding="application/x-tex">q = \frac{\sigma_s^2}{\sigma_g^2}</annotation></semantics></math>
is the proportion of additive genetic variance definitively explained by
the markers, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>
is the trait heritability.</p>
<p>This equation demonstrates the <strong>â€œLMSI Paradoxâ€</strong>
mathematically defined by Knapp (1998): as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>
trends toward zero,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>Î»</mi><mi>M</mi></msub><annotation encoding="application/x-tex">\lambda_M</annotation></semantics></math>
approaches infinity. Thus, the massive relative advantage of
marker-assisted selection over phenotypic selection occurs explicitly
when phenotypic trait heritability is extremely low, and the markers
successfully capture a massive fraction of genetic variance. Otherwise,
the LMSI offers diminishing returns and can fix QTLs at an overly fast
rate.</p>
</div>
<div class="section level3">
<h3 id="statistical-properties-of-molecular-indices">3.3 Statistical Properties of Molecular Indices<a class="anchor" aria-label="anchor" href="#statistical-properties-of-molecular-indices"></a>
</h3>
<p>Assuming
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and the index
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>I</mi><mi>M</mi></msub><annotation encoding="application/x-tex">I_M</annotation></semantics></math>
share a bivariate joint normal distribution, the statistical properties
mirror the standard phenotypic LPSI:</p>
<ol style="list-style-type: decimal">
<li>The variance of the index equals the covariance between the net
genetic merit and the index:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>Ïƒ</mi><msub><mi>I</mi><mi>M</mi></msub><mn>2</mn></msubsup><mo>=</mo><msub><mi>Ïƒ</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>M</mi></msub></mrow></msub></mrow><annotation encoding="application/x-tex">\sigma_{I_M}^2=\sigma_{HI_M}</annotation></semantics></math>.</li>
<li>The maximized correlation (Accuracy) is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>M</mi></msub></mrow></msub><mo>=</mo><mfrac><msub><mi>Ïƒ</mi><msub><mi>I</mi><mi>M</mi></msub></msub><msub><mi>Ïƒ</mi><mi>H</mi></msub></mfrac></mrow><annotation encoding="application/x-tex">\rho_{HI_M}=\frac{\sigma_{I_M}}{\sigma_H}</annotation></semantics></math>.</li>
<li>The Variance of the Predicted Error
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>V</mi><mi>a</mi><mi>r</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>H</mi><mo>âˆ’</mo><msub><mi>I</mi><mi>M</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>âˆ’</mo><msubsup><mi>Ï</mi><mrow><mi>H</mi><msub><mi>I</mi><mi>M</mi></msub></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><msubsup><mi>Ïƒ</mi><mi>H</mi><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">Var\left(H-{I}_M\right)=\left(1-{\rho}_{HI_M}^2\right){\sigma}_H^2</annotation></semantics></math>
is uniquely minimized.</li>
</ol>
<p>Because molecular indices incorporate <em>both</em> the phenotypic
and genotypic marker data, the LMSI accuracy will technically always be
higher than the standard LPSI accuracy.</p>
</div>
<div class="section level3">
<h3 id="advanced-matrix-estimation-methods">3.4 Advanced Matrix Estimation Methods<a class="anchor" aria-label="anchor" href="#advanced-matrix-estimation-methods"></a>
</h3>
<p>While the <code><a href="../reference/gw_lmsi.html">gw_lmsi()</a></code> and <code><a href="../reference/lmsi.html">lmsi()</a></code> functions in
the <code>selection.index</code> package abstract away the underlying
algebra, molecular index computation routinely suffers from the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>â‰«</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">p \gg n</annotation></semantics></math>
covariance problem (vastly more markers than recorded individuals).</p>
<p>Estimating the marker score variances
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ’</mi><annotation encoding="application/x-tex">\mathbf{S}</annotation></semantics></math>)
and the massive
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğŒ</mi><annotation encoding="application/x-tex">\mathbf{M}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ–</mi><annotation encoding="application/x-tex">\mathbf{W}</annotation></semantics></math>
covariance matrices using least-squares often leads to: 1. Marker
covariance values erroneously overestimating the true genetic covariance
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ‚</mi><annotation encoding="application/x-tex">\mathbf{C}</annotation></semantics></math>).
2. Singular, non-positive definite matrices where classical inverses
fail.</p>
<p>The standard solution involves regularization techniques (like the
Ridge penalty <code>lambda</code> built into <code><a href="../reference/gw_lmsi.html">gw_lmsi()</a></code>),
ensuring matrix invertibility at the cost of slight coefficient
bias.</p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The <code>selection.index</code> package extends base index selection
directly into the molecular realm. The <strong>LMSI</strong>
(<code><a href="../reference/lmsi.html">lmsi()</a></code>) allows breeders to leverage an established
pre-calculated marker scorings regression model. The
<strong>GW-LMSI</strong> (<code><a href="../reference/gw_lmsi.html">gw_lmsi()</a></code>) embraces the Genomic
Selection paradigm by swallowing all marker and phenotypic data jointly
in a single predictive framework, utilizing ridge regularization to
bypass classical
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>â‰«</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">p \gg n</annotation></semantics></math>
multicollinearity obstacles.</p>
<p>When implemented on traits exhibiting low heritability, these
marker-assisted pipelines can drastically outcompete standard phenotypic
indices in maximizing Expected Genetic Gain per selection cycle.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zankrut Goyani.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
