<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>The Linear Phenotypic Selection Index Theory â€¢ selection.index</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The Linear Phenotypic Selection Index Theory">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">selection.index</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/constrained-linear-phenotypic-selection-indices.html">Constrained Linear Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/Examples.html">Data Analysis with selection.index</a></li>
    <li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started with selection.index</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-selection-index.html">The Linear Phenotypic Selection Index Theory</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zankrut20/selection.index/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>The Linear Phenotypic Selection Index Theory</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zankrut20/selection.index/blob/feature/cpp-backend/vignettes/linear-phenotypic-selection-index.Rmd" class="external-link"><code>vignettes/linear-phenotypic-selection-index.Rmd</code></a></small>
      <div class="d-none name"><code>linear-phenotypic-selection-index.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In plant and animal breeding, quantitative traits (QTs) are
expressions of genes distributed across the genome interacting with the
environment. The phenotypic value of QTs
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>y</mi><annotation encoding="application/x-tex">y</annotation></semantics></math>)
can be systematically partitioned into a genotypic component
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>g</mi><annotation encoding="application/x-tex">g</annotation></semantics></math>)
and an environmental component
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>e</mi><annotation encoding="application/x-tex">e</annotation></semantics></math>):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>y</mi><mo>=</mo><mi>g</mi><mo>+</mo><mi>e</mi></mrow><annotation encoding="application/x-tex"> y = g + e </annotation></semantics></math></p>
<p>The primary goal in breeding is to maximize an individualâ€™s
<strong>net genetic merit</strong>. The net genetic merit
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>)
is a linear combination of the unobservable true breeding values
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ </mi><annotation encoding="application/x-tex">\mathbf{g}</annotation></semantics></math>)
weighted by their respective economic values
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ°</mi><annotation encoding="application/x-tex">\mathbf{w}</annotation></semantics></math>):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>H</mi><mo>=</mo><msup><mi>ğ°</mi><mi>â€²</mi></msup><mi>ğ </mi></mrow><annotation encoding="application/x-tex"> H = {\mathbf{w}}^{\prime}\mathbf{g} </annotation></semantics></math></p>
<p>Because the net genetic merit is unobservable in field trials,
breeders construct a <strong>Linear Phenotypic Selection Index
(LPSI)</strong> to predict it. The LPSI
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>)
is a linear combination of the observable and optimally weighted
phenotypic trait values
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ²</mi><annotation encoding="application/x-tex">\mathbf{y}</annotation></semantics></math>)
adjusted by index coefficients
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ›</mi><annotation encoding="application/x-tex">\mathbf{b}</annotation></semantics></math>):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>I</mi><mo>=</mo><msup><mi>ğ›</mi><mi>â€²</mi></msup><mi>ğ²</mi></mrow><annotation encoding="application/x-tex"> I = {\mathbf{b}}^{\prime}\mathbf{y} </annotation></semantics></math></p>
<p>The objective of the LPSI is to predict the net genetic merit and
maximize the multi-trait selection response.</p>
</div>
<div class="section level2">
<h2 id="optimizing-the-lpsi">Optimizing the LPSI<a class="anchor" aria-label="anchor" href="#optimizing-the-lpsi"></a>
</h2>
<p>To identify the optimal parents for the next selection cycle, the
correlation between the net genetic merit
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>)
and the LPSI
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>)
must be maximized. The vector
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ›</mi><annotation encoding="application/x-tex">\mathbf{b}</annotation></semantics></math>
that simultaneously minimizes the mean squared difference between
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>
and perfectly maximizes this correlation is mathematically derived
as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ›</mi><mo>=</mo><msup><mi>ğ</mi><mrow><mo>âˆ’</mo><mn>1</mn></mrow></msup><mrow><mi>ğ†</mi><mi>ğ°</mi></mrow></mrow><annotation encoding="application/x-tex"> \mathbf{b} = {\mathbf{P}}^{-1}\mathbf{Gw} </annotation></semantics></math></p>
<p>where: *
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>
is the phenotypic variance-covariance matrix. *
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ†</mi><annotation encoding="application/x-tex">\mathbf{G}</annotation></semantics></math>
is the genotypic variance-covariance matrix. *
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ°</mi><annotation encoding="application/x-tex">\mathbf{w}</annotation></semantics></math>
is the vector of economic weights defining relative trait
importance.</p>
<p>Once these optimal coefficients are derived, we can evaluate two
fundamental parameters:</p>
<ol style="list-style-type: decimal">
<li><p><strong>The Maximized Selection Response
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>R</mi><mi>I</mi></msub><annotation encoding="application/x-tex">R_I</annotation></semantics></math>)</strong>:
The expected mean improvement in the net genetic merit due to indirect
selection on the index.
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><mi>I</mi></msub><mo>=</mo><msub><mi>k</mi><mi>I</mi></msub><msqrt><mrow><msup><mi>ğ›</mi><mi>â€²</mi></msup><mrow><mi>ğ</mi><mi>ğ›</mi></mrow></mrow></msqrt></mrow><annotation encoding="application/x-tex"> {R}_I = {k}_I\sqrt{{\mathbf{b}}^{\prime}\mathbf{Pb}} </annotation></semantics></math></p></li>
<li><p><strong>The Expected Genetic Gain Per Trait
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ„</mi><annotation encoding="application/x-tex">\mathbf{E}</annotation></semantics></math>)</strong>:
The multi-trait selection response broken down per individual trait.
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ„</mi><mo>=</mo><msub><mi>k</mi><mi>I</mi></msub><mfrac><mrow><mi>ğ†</mi><mi>ğ›</mi></mrow><msub><mi>Ïƒ</mi><mi>I</mi></msub></mfrac></mrow><annotation encoding="application/x-tex"> \mathbf{E} = {k}_I\frac{\mathbf{Gb}}{\sigma_I} </annotation></semantics></math></p></li>
</ol>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>k</mi><mi>I</mi></msub><annotation encoding="application/x-tex">k_I</annotation></semantics></math>
is the standardized selection intensity and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>Ïƒ</mi><mi>I</mi></msub><annotation encoding="application/x-tex">\sigma_I</annotation></semantics></math>
is the standard deviation of the index score variance.</p>
</div>
<div class="section level2">
<h2 id="practical-implementation-in-r">Practical Implementation in R<a class="anchor" aria-label="anchor" href="#practical-implementation-in-r"></a>
</h2>
<p>We can seamlessly translate this text theory into rigorous
statistical practice using the <code>selection.index</code> package. We
will utilize the built-in synthetic datasets: <code>maize_pheno</code>
(containing multi-environment phenotypic records for 100 genotypes) and
<code>maize_geno</code> (500 SNP markers).</p>
<div class="section level3">
<h3 id="estimating-covariance-matrices">1. Estimating Covariance Matrices<a class="anchor" aria-label="anchor" href="#estimating-covariance-matrices"></a>
</h3>
<p>First, we estimate the genotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ†</mi><annotation encoding="application/x-tex">\mathbf{G}</annotation></semantics></math>)
and phenotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>)
variance-covariance matrices from our raw phenotypic dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zankrut20/selection.index" class="external-link">selection.index</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the synthetic phenotypic multi-environment dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_pheno"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># In maize_pheno: Traits are columns 4:6.</span></span>
<span><span class="co"># Genotypes are in column 1, and Block/Replication is in column 3.</span></span>
<span><span class="va">gmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gen_varcov.html">gen_varcov</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>, genotypes <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, replication <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phen_varcov.html">phen_varcov</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>, genotypes <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, replication <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="defining-economic-weights">2. Defining Economic Weights<a class="anchor" aria-label="anchor" href="#defining-economic-weights"></a>
</h3>
<p>Next, we establish the relative economic priority of each trait.
Economic weights
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ°</mi><annotation encoding="application/x-tex">\mathbf{w}</annotation></semantics></math>)
explicitly define our strategic breeding objectives.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define the economic weights for the 3 continuous traits</span></span>
<span><span class="co"># (e.g., Yield, PlantHeight, DaysToMaturity)</span></span>
<span><span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="op">-</span><span class="fl">5</span>, <span class="op">-</span><span class="fl">5</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="calculating-the-lpsi">3. Calculating the LPSI<a class="anchor" aria-label="anchor" href="#calculating-the-lpsi"></a>
</h3>
<p>With the covariance matrices and economic weights specified, we
integrate them into the primary <code><a href="../reference/lpsi.html">lpsi()</a></code> function, which
evaluates the combinatorial multi-trait selection indices
efficiently.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the Optimal Combinatorial Linear Phenotypic Selection Index (LPSI)</span></span>
<span><span class="va">index_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lpsi.html">lpsi</a></span><span class="op">(</span></span>
<span>    ncomb <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    pmat <span class="op">=</span> <span class="va">pmat</span>,</span>
<span>    gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>    wmat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">weights</span><span class="op">)</span>,</span>
<span>    wcol <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="evaluating-outcomes-and-selecting-genotypes">4. Evaluating Outcomes and Selecting Genotypes<a class="anchor" aria-label="anchor" href="#evaluating-outcomes-and-selecting-genotypes"></a>
</h3>
<p>Finally, we evaluate the theoretical gains. The <code><a href="../reference/lpsi.html">lpsi()</a></code>
function returns a structured data frame containing the theoretical
selection response
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>R</mi><mi>I</mi></msub><annotation encoding="application/x-tex">R_I</annotation></semantics></math>)
and other parameter estimates for all requested trait combinations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View the top combinatorial indices, including their selection response (R_A)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">index_results</span><span class="op">)</span></span>
<span><span class="co">#&gt;        ID      b.1      b.2     b.3      GA       PRE Delta_G     rHI      hI2</span></span>
<span><span class="co">#&gt; 1 1, 2, 3 2364.671 9453.933 1012156 6863519 686351898 6863519 42.0899 1771.794</span></span>
<span><span class="co">#&gt;   Rank</span></span>
<span><span class="co">#&gt; 1    1</span></span>
<span></span>
<span><span class="co"># Extract the phenotypic selection scores to strategically rank the parental candidates</span></span>
<span><span class="co"># using the top evaluated combinatorial index</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_selection_score.html">predict_selection_score</a></span><span class="op">(</span></span>
<span>    <span class="va">index_results</span>,</span>
<span>    data <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>,</span>
<span>    genotypes <span class="op">=</span> <span class="va">maize_pheno</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the top performing candidates designated for the next breeding cycle</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genotypes   I_1_2_3 I_1_2_3_Rank</span></span>
<span><span class="co">#&gt; 1      G001 138639943           78</span></span>
<span><span class="co">#&gt; 2      G002 138393284           83</span></span>
<span><span class="co">#&gt; 3      G003 138260546           88</span></span>
<span><span class="co">#&gt; 4      G004 139442740           46</span></span>
<span><span class="co">#&gt; 5      G005 139319528           51</span></span>
<span><span class="co">#&gt; 6      G006 139661606           36</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extension-linear-marker-selection-index">5. Extension: Linear Marker Selection Index<a class="anchor" aria-label="anchor" href="#extension-linear-marker-selection-index"></a>
</h3>
<p>The classical linear selection index theories seamlessly extend to
marker-assisted genomic selection. If you have genome-wide marker
profiles for your genotypes, you can incorporate them to estimate the
Linear Marker Selection Index (LMSI).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the associated synthetic genomic dataset (500 SNPs for the 100 genotypes)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"maize_geno"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate the marker-assisted index combining our matrices and raw SNP profiles</span></span>
<span><span class="va">marker_index_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lmsi.html">lmsi</a></span><span class="op">(</span></span>
<span>    pmat <span class="op">=</span> <span class="va">pmat</span>,</span>
<span>    gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>    marker_scores <span class="op">=</span> <span class="va">maize_geno</span>,</span>
<span>    wmat <span class="op">=</span> <span class="va">weights</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">marker_index_results</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="the-base-index-and-index-efficiency">6. The Base Index and Index Efficiency<a class="anchor" aria-label="anchor" href="#the-base-index-and-index-efficiency"></a>
</h3>
<p>In scenarios where the phenotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>)
and genotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ†</mi><annotation encoding="application/x-tex">\mathbf{G}</annotation></semantics></math>)
matrices are poorly estimated (e.g., due to limited data), the true
optimal coefficients
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ›</mi><annotation encoding="application/x-tex">\mathbf{b}</annotation></semantics></math>)
can be systematically biased. The <strong>Base Index</strong> provides a
robust, non-optimized alternative where coefficients are set strictly
equal to the fixed economic weights
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>I</mi><mi>B</mi></msub><mo>=</mo><mi>ğ°</mi><mi>â€²</mi><mi>ğ²</mi></mrow><annotation encoding="application/x-tex">I_B = \mathbf{w}'\mathbf{y}</annotation></semantics></math>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the Base Index and automatically compare its efficiency to the LPSI</span></span>
<span><span class="va">base_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/base_index.html">base_index</a></span><span class="op">(</span></span>
<span>    pmat <span class="op">=</span> <span class="va">pmat</span>,</span>
<span>    gmat <span class="op">=</span> <span class="va">gmat</span>,</span>
<span>    wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>    compare_to_lpsi <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Observe the expected genetic gains and efficiency comparison</span></span>
<span><span class="va">base_results</span><span class="op">$</span><span class="va">summary</span></span>
<span><span class="co">#&gt;   b.1 b.2 b.3      GA       PRE  Delta_G      hI2     rHI</span></span>
<span><span class="co">#&gt; 1  10  -5  -5 1824121 182412094 14577.53 125.1324 11.1863</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="heritability-of-the-lpsi">7. Heritability of the LPSI<a class="anchor" aria-label="anchor" href="#heritability-of-the-lpsi"></a>
</h3>
<p>The theory demonstrates that the correlation between the net genetic
merit
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>H</mi><annotation encoding="application/x-tex">H</annotation></semantics></math>)
and the expected index
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>I</mi><annotation encoding="application/x-tex">I</annotation></semantics></math>)
differs from the traditional index heritability mathematically
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>h</mi><mi>I</mi><mn>2</mn></msubsup><mo>â‰ </mo><msubsup><mi>Ï</mi><mrow><mi>H</mi><mi>I</mi></mrow><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">h^2_I \neq \rho^2_{HI}</annotation></semantics></math>).
The <code><a href="../reference/lpsi.html">lpsi()</a></code> function intrinsically estimates both of these
fundamental statistics:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract the top combinatorial index results</span></span>
<span><span class="va">top_index</span> <span class="op">&lt;-</span> <span class="va">index_results</span><span class="op">[</span><span class="fl">1</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># h^2_I: Heritability of the optimal index</span></span>
<span><span class="va">top_index</span><span class="op">$</span><span class="va">hI2</span></span>
<span><span class="co">#&gt; [1] 1771.794</span></span>
<span></span>
<span><span class="co"># \rho_HI: Correlation between the LPSI and the true underlying Net Genetic Merit</span></span>
<span><span class="va">top_index</span><span class="op">$</span><span class="va">rHI</span></span>
<span><span class="co">#&gt; [1] 42.0899</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zankrut Goyani.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
