<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Multistage Phenotypic Selection Indices â€¢ selection.index</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><script src="../extra.js"></script><meta property="og:title" content="Multistage Phenotypic Selection Indices">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">selection.index</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/getting-started.html">Getting Started with selection.index</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-selection-index.html">The Linear Phenotypic Selection Index Theory</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-phenotypic-selection-indices.html">Constrained Linear Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-marker-selection-indices.html">Linear Marker and Genome-Wide Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-selection-indices.html">Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/constrained-linear-genomic-selection-indices.html">Constrained Linear Genomic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/linear-phenotypic-eigen-selection-index-methods.html">Linear Phenotypic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/linear-genomic-eigen-selection-index-methods.html">Linear Genomic Eigen Selection Index Methods</a></li>
    <li><a class="dropdown-item" href="../articles/multistage-linear-selection-indices.html">Multistage Phenotypic Selection Indices</a></li>
    <li><a class="dropdown-item" href="../articles/stochastic-simulation-linear-phenotypic-indices.html">Stochastic Simulation of Linear Phenotypic Selection Indices</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zankrut20/selection.index/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Multistage Phenotypic Selection Indices</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zankrut20/selection.index/blob/master/vignettes/multistage-linear-selection-indices.Rmd" class="external-link"><code>vignettes/multistage-linear-selection-indices.Rmd</code></a></small>
      <div class="d-none name"><code>multistage-linear-selection-indices.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In crop and animal breeding programs, selection is rarely performed
in a single stage. Breeders usually evaluate multiple traits across
different stages of testing, successively discarding inferior genotypes
and advancing superior ones. This multistage approach requires selection
indices to account for the changes in variances and covariances induced
by selection at prior stages.</p>
<p>Chapter 9 of the <code>selection.index</code> package focuses on the
mathematical formulation and practical application of Multistage Linear
Selection Indices. We introduce indices that properly adjust for prior
selection using the Cochran (1951) and Cunningham (1975) method,
calculating corrected covariance matrices to appropriately predict
subsequent selection responses. This vignette will demonstrate six
multistage indices utilizing phenotypic and genomic estimated breeding
values (GEBVs).</p>
<p>We will use the synthetic maize phenotypic and genotypic datasets
(<code>maize_pheno</code> and <code>maize_geno</code>) to illustrate
these complex functions. Let us first prepare the covariance
matrices.</p>
<div class="section level3">
<h3 id="data-preparation">Data Preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h3>
<p>For our examples, we will evaluate 3 quantitative traits from the
dataset: Yield, PlantHeight, and DaysToMaturity. We assume that Stage 1
selection evaluates the first two traits (Yield, PlantHeight), and Stage
2 evaluates all 3 traits (adding DaysToMaturity).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zankrut20/selection.index" class="external-link">selection.index</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Estimate phenotypic and genotypic covariance matrices for the 3 traits</span></span>
<span><span class="co"># The traits are Yield, PlantHeight, DaysToMaturity</span></span>
<span><span class="va">traits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Yield"</span>, <span class="st">"PlantHeight"</span>, <span class="st">"DaysToMaturity"</span><span class="op">)</span></span>
<span><span class="va">pmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/phen_varcov.html">phen_varcov</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Environment</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span><span class="op">)</span></span>
<span><span class="va">gmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gen_varcov.html">gen_varcov</a></span><span class="op">(</span><span class="va">maize_pheno</span><span class="op">[</span>, <span class="va">traits</span><span class="op">]</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Environment</span>, <span class="va">maize_pheno</span><span class="op">$</span><span class="va">Genotype</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Matrix limits for Stage 1 (Traits 1 to 2)</span></span>
<span><span class="va">P1</span> <span class="op">&lt;-</span> <span class="va">pmat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">G1</span> <span class="op">&lt;-</span> <span class="va">gmat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Complete Matrices for Stage 2</span></span>
<span><span class="va">P</span> <span class="op">&lt;-</span> <span class="va">pmat</span></span>
<span><span class="va">C</span> <span class="op">&lt;-</span> <span class="va">gmat</span></span>
<span></span>
<span><span class="co"># Economic weights for the 3 traits</span></span>
<span><span class="va">weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="op">-</span><span class="fl">5</span>, <span class="op">-</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="multistage-linear-phenotypic-selection-index-mlpsi">1. Multistage Linear Phenotypic Selection Index (MLPSI)<a class="anchor" aria-label="anchor" href="#multistage-linear-phenotypic-selection-index-mlpsi"></a>
</h2>
<p>The Multistage Linear Phenotypic Selection Index accounts for changes
in phenotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>)
and genotypic
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ‚</mi><annotation encoding="application/x-tex">\mathbf{C}</annotation></semantics></math>)
covariance matrices due to previous selection cycles.</p>
<p>At Stage 1, the index coefficients are computed just as in the
standard Smith-Hazel index:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ›</mi><mn>1</mn></msub><mo>=</mo><msubsup><mi>ğ</mi><mn>1</mn><mrow><mo>âˆ’</mo><mn>1</mn></mrow></msubsup><msub><mi>ğ†</mi><mn>1</mn></msub><msub><mi>ğ°</mi><mn>1</mn></msub></mrow><annotation encoding="application/x-tex"> \mathbf{b}_1 = \mathbf{P}_1^{-1}\mathbf{G}_1\mathbf{w}_1 </annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ°</mi><mn>1</mn></msub><annotation encoding="application/x-tex">\mathbf{w}_1</annotation></semantics></math>
contains the economic weights for Stage 1 traits.</p>
<p>At Stage 2, the coefficients for the entire set of traits are:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ›</mi><mn>2</mn></msub><mo>=</mo><msup><mi>ğ</mi><mrow><mo>âˆ’</mo><mn>1</mn></mrow></msup><mi>ğ‚</mi><mi>ğ°</mi></mrow><annotation encoding="application/x-tex"> \mathbf{b}_2 = \mathbf{P}^{-1}\mathbf{C}\mathbf{w} </annotation></semantics></math></p>
<p>However, due to selection at stage 1, the covariances
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ‚</mi><annotation encoding="application/x-tex">\mathbf{C}</annotation></semantics></math>
are adjusted for stage 2 evaluation:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>ğ</mi><mo>*</mo></msup><mo>=</mo><mi>ğ</mi><mo>âˆ’</mo><mi>u</mi><mfrac><mrow><mi>C</mi><mi>o</mi><mi>v</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>ğ²</mi><mo>,</mo><msub><mi>ğ±</mi><mn>1</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><msub><mi>ğ›</mi><mn>1</mn></msub><msub><mi>ğ›</mi><mn>1</mn></msub><mi>â€²</mi><mi>C</mi><mi>o</mi><mi>v</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>ğ±</mi><mn>1</mn></msub><mo>,</mo><mi>ğ²</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><msub><mi>ğ›</mi><mn>1</mn></msub><mi>â€²</mi><msub><mi>ğ</mi><mn>1</mn></msub><msub><mi>ğ›</mi><mn>1</mn></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex"> \mathbf{P}^* = \mathbf{P} - u \frac{Cov(\mathbf{y},\mathbf{x}_1)\mathbf{b}_1\mathbf{b}_1'Cov(\mathbf{x}_1,\mathbf{y})}{\mathbf{b}_1'\mathbf{P}_1\mathbf{b}_1} </annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>ğ‚</mi><mo>*</mo></msup><mo>=</mo><mi>ğ‚</mi><mo>âˆ’</mo><mi>u</mi><mfrac><mrow><msub><mi>ğ†</mi><mn>1</mn></msub><mi>â€²</mi><msub><mi>ğ›</mi><mn>1</mn></msub><msub><mi>ğ›</mi><mn>1</mn></msub><mi>â€²</mi><msub><mi>ğ†</mi><mn>1</mn></msub></mrow><mrow><msub><mi>ğ›</mi><mn>1</mn></msub><mi>â€²</mi><msub><mi>ğ</mi><mn>1</mn></msub><msub><mi>ğ›</mi><mn>1</mn></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex"> \mathbf{C}^* = \mathbf{C} - u \frac{\mathbf{G}_1'\mathbf{b}_1\mathbf{b}_1'\mathbf{G}_1}{\mathbf{b}_1'\mathbf{P}_1\mathbf{b}_1} </annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>u</mi><mo>=</mo><msub><mi>k</mi><mn>1</mn></msub><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>k</mi><mn>1</mn></msub><mo>âˆ’</mo><mi>Ï„</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">u = k_1(k_1 - \tau)</annotation></semantics></math>
calculates the effect of selection based on the standardized truncation
point
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>Ï„</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
and selection intensity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>k</mi><mn>1</mn></msub><annotation encoding="application/x-tex">k_1</annotation></semantics></math>.</p>
<p>The function <code>mlpsi</code> simultaneously performs adjustments
and metric estimations for both stages.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We apply a selection proportion of 10% (0.10) per stage.</span></span>
<span><span class="va">mlpsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mlpsi.html">mlpsi</a></span><span class="op">(</span></span>
<span>  P1 <span class="op">=</span> <span class="va">P1</span>, P <span class="op">=</span> <span class="va">P</span>, G1 <span class="op">=</span> <span class="va">G1</span>, C <span class="op">=</span> <span class="va">C</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .index_correlation(b1, b2, P1, P, stage1_indices): Invalid variance</span></span>
<span><span class="co">#&gt; for correlation calculation.</span></span>
<span><span class="co">#&gt; Warning in .adjust_phenotypic_matrix(P, P1, b1, k1, tau, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span><span class="co">#&gt; Warning in .adjust_genotypic_matrix(C, G1, b1, k1, tau, P1, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span></span>
<span><span class="co"># Stage 1 metrics</span></span>
<span><span class="va">mlpsi_res</span><span class="op">$</span><span class="va">summary_stage1</span></span>
<span><span class="co">#&gt;   Stage       Trait             b  E</span></span>
<span><span class="co">#&gt; 1     1       Yield -527740915826 NA</span></span>
<span><span class="co">#&gt; 2     1 PlantHeight 4066491692882 NA</span></span>
<span></span>
<span><span class="co"># Stage 2 metrics</span></span>
<span><span class="va">mlpsi_res</span><span class="op">$</span><span class="va">summary_stage2</span></span>
<span><span class="co">#&gt;   Stage          Trait             b         E</span></span>
<span><span class="co">#&gt; 1     2          Yield -1.167617e+12 857097221</span></span>
<span><span class="co">#&gt; 2     2    PlantHeight  1.254688e+13  22420339</span></span>
<span><span class="co">#&gt; 3     2 DaysToMaturity  7.244343e+13  12421006</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="multistage-restricted-linear-phenotypic-selection-index-mrlpsi">2. Multistage Restricted Linear Phenotypic Selection Index
(MRLPSI)<a class="anchor" aria-label="anchor" href="#multistage-restricted-linear-phenotypic-selection-index-mrlpsi"></a>
</h2>
<p>The MRLPSI method applies when the breeder aims to maintain one or
more quantitative traits without change over the multistage evaluation
(e.g., maintaining constant <code>PlantHeight</code> while optimizing
other variables).</p>
<p>The restricted coefficient vectors for Stage 1 and Stage 2 are
defined as:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ›</mi><msub><mi>R</mi><mn>1</mn></msub></msub><mo>=</mo><msub><mi>ğŠ</mi><mn>1</mn></msub><msub><mi>ğ›</mi><mn>1</mn></msub></mrow><annotation encoding="application/x-tex"> \mathbf{b}_{R_1} = \mathbf{K}_1 \mathbf{b}_1 </annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğ›</mi><msub><mi>R</mi><mn>2</mn></msub></msub><mo>=</mo><msub><mi>ğŠ</mi><mn>2</mn></msub><msub><mi>ğ›</mi><mn>2</mn></msub></mrow><annotation encoding="application/x-tex"> \mathbf{b}_{R_2} = \mathbf{K}_2 \mathbf{b}_2 </annotation></semantics></math></p>
<p>Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğŠ</mi><mn>1</mn></msub><mo>=</mo><msub><mi>ğˆ</mi><mn>1</mn></msub><mo>âˆ’</mo><msub><mi>ğ</mi><mn>1</mn></msub></mrow><annotation encoding="application/x-tex">\mathbf{K}_1 = \mathbf{I}_1 - \mathbf{Q}_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ğŠ</mi><mn>2</mn></msub><mo>=</mo><msub><mi>ğˆ</mi><mn>2</mn></msub><mo>âˆ’</mo><msub><mi>ğ</mi><mn>2</mn></msub></mrow><annotation encoding="application/x-tex">\mathbf{K}_2 = \mathbf{I}_2 - \mathbf{Q}_2</annotation></semantics></math>
are matrices that impose zero genetic gain vectors, derived from the
constraint matrices
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ‚</mi><mn>1</mn></msub><annotation encoding="application/x-tex">\mathbf{C}_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ‚</mi><mn>2</mn></msub><annotation encoding="application/x-tex">\mathbf{C}_2</annotation></semantics></math>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We constrain PlantHeight (Trait 2) at Stage 1</span></span>
<span><span class="va">C1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">C1</span><span class="op">[</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="co"># We constrain PlantHeight (Trait 2) at Stage 2</span></span>
<span><span class="va">C2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">C2</span><span class="op">[</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">mrlpsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mrlpsi.html">mrlpsi</a></span><span class="op">(</span></span>
<span>  P1 <span class="op">=</span> <span class="va">P1</span>, P <span class="op">=</span> <span class="va">P</span>, G1 <span class="op">=</span> <span class="va">G1</span>, C <span class="op">=</span> <span class="va">C</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  C1 <span class="op">=</span> <span class="va">C1</span>, C2 <span class="op">=</span> <span class="va">C2</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .index_correlation(b_R1, b_R2, P1, P, stage1_indices): Invalid</span></span>
<span><span class="co">#&gt; variance for correlation calculation.</span></span>
<span><span class="co">#&gt; Warning in .adjust_phenotypic_matrix(P, P1, b_R1, k1, tau, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span><span class="co">#&gt; Warning in .adjust_genotypic_matrix(C, G1, b_R1, k1, tau, P1, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span></span>
<span><span class="co"># Observe that Expected Gain (E) for PlantHeight is approximately 0</span></span>
<span><span class="va">mrlpsi_res</span><span class="op">$</span><span class="va">summary_stage1</span></span>
<span><span class="co">#&gt;   Stage       Trait           b_R  E</span></span>
<span><span class="co">#&gt; 1     1       Yield -529172810900 NA</span></span>
<span><span class="co">#&gt; 2     1 PlantHeight 4077525117142 NA</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="multistage-predetermined-proportional-gain-lpsi-mppg-lpsi">3. Multistage Predetermined Proportional Gain LPSI (MPPG-LPSI)<a class="anchor" aria-label="anchor" href="#multistage-predetermined-proportional-gain-lpsi-mppg-lpsi"></a>
</h2>
<p>Unlike MRLPSI which imposes zero genetic gain bounds, MPPG-LPSI
forces proportional changes mapped by the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ</mi><mn>1</mn></msub><annotation encoding="application/x-tex">\mathbf{d}_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>ğ</mi><mn>2</mn></msub><annotation encoding="application/x-tex">\mathbf{d}_2</annotation></semantics></math>
restricted-difference vectors. At Stage
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>,
the difference vector creates an updated target trajectory.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Target specific proportional gains</span></span>
<span><span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span> <span class="co"># Yield gains twice as much as PlantHeight at stage 1</span></span>
<span><span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span> <span class="co"># Desired proportions at stage 2</span></span>
<span></span>
<span><span class="va">mppg_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mppg_lpsi.html">mppg_lpsi</a></span><span class="op">(</span></span>
<span>  P1 <span class="op">=</span> <span class="va">P1</span>, P <span class="op">=</span> <span class="va">P</span>, G1 <span class="op">=</span> <span class="va">G1</span>, C <span class="op">=</span> <span class="va">C</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  d1 <span class="op">=</span> <span class="va">d1</span>, d2 <span class="op">=</span> <span class="va">d2</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .index_correlation(b_M1, b_M2, P1, P, stage1_indices): Invalid</span></span>
<span><span class="co">#&gt; variance for correlation calculation.</span></span>
<span><span class="co">#&gt; Warning in .adjust_phenotypic_matrix(P, P1, b_M1, k1, tau, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span><span class="co">#&gt; Warning in .adjust_genotypic_matrix(C, G1, b_M1, k1, tau, P1, stage1_indices):</span></span>
<span><span class="co">#&gt; Invalid variance at stage 1 (b1'P1b1 &lt;= 0). Returning unadjusted matrix.</span></span>
<span></span>
<span><span class="co"># Observe the Expected Gain (E) in the resulting summary stats aligns with d1 proportions</span></span>
<span><span class="va">mppg_res</span><span class="op">$</span><span class="va">summary_stage1</span></span>
<span><span class="co">#&gt;   Stage       Trait           b_M d  E Ratio</span></span>
<span><span class="co">#&gt; 1     1       Yield -522178847351 2 NA    NA</span></span>
<span><span class="co">#&gt; 2     1 PlantHeight 4023633342185 1 NA    NA</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="multistage-linear-genomic-selection-index-mlgsi">4. Multistage Linear Genomic Selection Index (MLGSI)<a class="anchor" aria-label="anchor" href="#multistage-linear-genomic-selection-index-mlgsi"></a>
</h2>
<p>In modern breeding, Genomic Estimated Breeding Values (GEBVs)
computed from whole-genome markers accelerate cyclical selection. For
the Multi-Stage Genomic framework, we replace phenotypic estimators with
GEBV matrix derivations:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğšª</mi><annotation encoding="application/x-tex">\mathbf{\Gamma}</annotation></semantics></math>
(GEBV variance-covariance) replaces
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ€</mi><annotation encoding="application/x-tex">\mathbf{A}</annotation></semantics></math>
(Covariance between GEBVs and true BVs) provides genomic mappings.</p>
<p>For illustrative purposes, we mock simulate the arrays via a
pseudo-reliability scaling of the genetic matrices:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">reliability</span> <span class="op">&lt;-</span> <span class="fl">0.7</span> <span class="co"># Simulated genomic prediction reliability</span></span>
<span></span>
<span><span class="va">Gamma1</span> <span class="op">&lt;-</span> <span class="va">reliability</span> <span class="op">*</span> <span class="va">G1</span></span>
<span><span class="va">Gamma</span> <span class="op">&lt;-</span> <span class="va">reliability</span> <span class="op">*</span> <span class="va">C</span></span>
<span><span class="va">A1</span> <span class="op">&lt;-</span> <span class="va">reliability</span> <span class="op">*</span> <span class="va">G1</span></span>
<span><span class="va">A</span> <span class="op">&lt;-</span> <span class="va">C</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="co"># n x n1 covariance mapping</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mlgsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mlgsi.html">mlgsi</a></span><span class="op">(</span></span>
<span>  Gamma1 <span class="op">=</span> <span class="va">Gamma1</span>, Gamma <span class="op">=</span> <span class="va">Gamma</span>, A1 <span class="op">=</span> <span class="va">A1</span>, A <span class="op">=</span> <span class="va">A</span>,</span>
<span>  C <span class="op">=</span> <span class="va">C</span>, G1 <span class="op">=</span> <span class="va">G1</span>, P1 <span class="op">=</span> <span class="va">P1</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in mlgsi(Gamma1 = Gamma1, Gamma = Gamma, A1 = A1, A = A, C = C, :</span></span>
<span><span class="co">#&gt; Invalid phenotypic variance at stage 1. Using unadjusted C.</span></span>
<span></span>
<span><span class="va">mlgsi_res</span><span class="op">$</span><span class="va">summary_stage1</span></span>
<span><span class="co">#&gt;   Stage       Trait beta         E</span></span>
<span><span class="co">#&gt; 1     1       Yield   10 4822.1921</span></span>
<span><span class="co">#&gt; 2     1 PlantHeight   -5  126.4395</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="multistage-restricted-genomic-selection-index-mrlgsi">5. Multistage Restricted Genomic Selection Index (MRLGSI)<a class="anchor" aria-label="anchor" href="#multistage-restricted-genomic-selection-index-mrlgsi"></a>
</h2>
<p>Similarly, traits can be biologically constrained in multiple
genome-assisted breeding cycles.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mrlgsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mrlgsi.html">mrlgsi</a></span><span class="op">(</span></span>
<span>  Gamma1 <span class="op">=</span> <span class="va">Gamma1</span>, Gamma <span class="op">=</span> <span class="va">Gamma</span>, A1 <span class="op">=</span> <span class="va">A1</span>, A <span class="op">=</span> <span class="va">A</span>,</span>
<span>  C <span class="op">=</span> <span class="va">C</span>, G1 <span class="op">=</span> <span class="va">G1</span>, P1 <span class="op">=</span> <span class="va">P1</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  C1 <span class="op">=</span> <span class="va">C1</span>, C2 <span class="op">=</span> <span class="va">C2</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">mrlgsi_res</span><span class="op">$</span><span class="va">summary_stage2</span></span>
<span><span class="co">#&gt;   Stage          Trait    beta_R  E</span></span>
<span><span class="co">#&gt; 1     2          Yield   10.0000 NA</span></span>
<span><span class="co">#&gt; 2     2    PlantHeight -380.2161 NA</span></span>
<span><span class="co">#&gt; 3     2 DaysToMaturity   -2.0000 NA</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="multistage-ppg-genomic-selection-index-mppg-lgsi">6. Multistage PPG Genomic Selection Index (MPPG-LGSI)<a class="anchor" aria-label="anchor" href="#multistage-ppg-genomic-selection-index-mppg-lgsi"></a>
</h2>
<p>The procedure calculates predetermined gains over multiple cycles
exclusively utilizing whole-genome predictions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mppg_lgsi_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mppg_lgsi.html">mppg_lgsi</a></span><span class="op">(</span></span>
<span>  Gamma1 <span class="op">=</span> <span class="va">Gamma1</span>, Gamma <span class="op">=</span> <span class="va">Gamma</span>, A1 <span class="op">=</span> <span class="va">A1</span>, A <span class="op">=</span> <span class="va">A</span>,</span>
<span>  C <span class="op">=</span> <span class="va">C</span>, G1 <span class="op">=</span> <span class="va">G1</span>, P1 <span class="op">=</span> <span class="va">P1</span>,</span>
<span>  wmat <span class="op">=</span> <span class="va">weights</span>,</span>
<span>  d1 <span class="op">=</span> <span class="va">d1</span>, d2 <span class="op">=</span> <span class="va">d2</span>,</span>
<span>  selection_proportion <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in mppg_lgsi(Gamma1 = Gamma1, Gamma = Gamma, A1 = A1, A = A, C = C, :</span></span>
<span><span class="co">#&gt; Invalid phenotypic variance at stage 1. Using unadjusted C.</span></span>
<span></span>
<span><span class="va">mppg_lgsi_res</span><span class="op">$</span><span class="va">summary_stage1</span></span>
<span><span class="co">#&gt;   Stage       Trait  beta_P d      E  Ratio</span></span>
<span><span class="co">#&gt; 1     1       Yield -0.4151 2 3.2116 1.6058</span></span>
<span><span class="co">#&gt; 2     1 PlantHeight 15.8301 1 1.6058 1.6058</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="statistical-properties">Statistical Properties<a class="anchor" aria-label="anchor" href="#statistical-properties"></a>
</h2>
<p>For all the multistage indices above (Phenotypic and Genomic), we
evaluate statistical properties to compare efficiency.</p>
<div class="section level4">
<h4 id="accuracy">Accuracy<a class="anchor" aria-label="anchor" href="#accuracy"></a>
</h4>
<p>The accuracy (or correlation between the index and true breeding
value) indicates the efficiency of the index:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï</mi><mi>H</mi></msub><mo>=</mo><mfrac><msub><mi>Ïƒ</mi><mrow><mi>H</mi><mo>,</mo><mi>I</mi></mrow></msub><mrow><msub><mi>Ïƒ</mi><mi>H</mi></msub><msub><mi>Ïƒ</mi><mi>I</mi></msub></mrow></mfrac><mo>=</mo><msqrt><mfrac><mrow><mi>ğ›</mi><mi>â€²</mi><mi>ğ</mi><mi>ğ›</mi></mrow><mrow><mi>ğ°</mi><mi>â€²</mi><mi>ğ‚</mi><mi>ğ°</mi></mrow></mfrac></msqrt></mrow><annotation encoding="application/x-tex"> \rho_{H} = \frac{\sigma_{H, I}}{\sigma_H \sigma_I} = \sqrt{ \frac{\mathbf{b}'\mathbf{P}\mathbf{b}}{\mathbf{w}'\mathbf{C}\mathbf{w}} } </annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ</mi><annotation encoding="application/x-tex">\mathbf{P}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ğ‚</mi><annotation encoding="application/x-tex">\mathbf{C}</annotation></semantics></math>
are substituted by their adjusted equivalents at Stage 2
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>ğ</mi><mo>*</mo></msup><annotation encoding="application/x-tex">\mathbf{P}^*</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>ğ‚</mi><mo>*</mo></msup><annotation encoding="application/x-tex">\mathbf{C}^*</annotation></semantics></math>
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>ğšª</mi><mo>*</mo></msup><annotation encoding="application/x-tex">\mathbf{\Gamma}^*</annotation></semantics></math>).</p>
</div>
<div class="section level4">
<h4 id="selection-response">Selection Response<a class="anchor" aria-label="anchor" href="#selection-response"></a>
</h4>
<p>The overall selection response generated by the index evaluates the
genetic superiority:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>R</mi><mo>=</mo><mi>k</mi><msub><mi>Ïƒ</mi><mi>I</mi></msub><mo>=</mo><mi>k</mi><msqrt><mrow><mi>ğ›</mi><mi>â€²</mi><mi>ğ</mi><mi>ğ›</mi></mrow></msqrt></mrow><annotation encoding="application/x-tex"> R = k \sigma_{I} = k \sqrt{\mathbf{b}'\mathbf{P}\mathbf{b}} </annotation></semantics></math>
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is the selection intensity.</p>
</div>
<div class="section level4">
<h4 id="expected-genetic-gain">Expected Genetic Gain<a class="anchor" aria-label="anchor" href="#expected-genetic-gain"></a>
</h4>
<p>The expected genetic gain per individual trait is given by the
vector:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ„</mi><mo>=</mo><mi>k</mi><mfrac><mrow><mi>ğ†</mi><mi>â€²</mi><mi>ğ›</mi></mrow><msub><mi>Ïƒ</mi><mi>I</mi></msub></mfrac><mo>=</mo><mi>k</mi><mfrac><mrow><mi>ğ†</mi><mi>â€²</mi><mi>ğ›</mi></mrow><msqrt><mrow><mi>ğ›</mi><mi>â€²</mi><mi>ğ</mi><mi>ğ›</mi></mrow></msqrt></mfrac></mrow><annotation encoding="application/x-tex"> \mathbf{E} = k \frac{\mathbf{G}'\mathbf{b}}{\sigma_I} = k \frac{\mathbf{G}'\mathbf{b}}{\sqrt{\mathbf{b}'\mathbf{P}\mathbf{b}}} </annotation></semantics></math>
For the Multistage Genomic Indices,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ†</mi><mi>â€²</mi></mrow><annotation encoding="application/x-tex">\mathbf{G}'</annotation></semantics></math>
is replaced by the marker association mapping matrix
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ğ€</mi><mi>â€²</mi></mrow><annotation encoding="application/x-tex">\mathbf{A}'</annotation></semantics></math>.</p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Multistage Selection Indices effectively manage breeding resources by
filtering inferior variants sequentially while accurately recalibrating
covariance parameters using genomic and phenotypic variables across
progressive evaluations. These analytical tools prevent severe variance
distortion over breeding cycles.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zankrut Goyani.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
